esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -0,0 +1,127 @@
+"""Provides an iris interface for unstructured regridding"""
+
+from numpy import ma
+import iris
+from iris.analysis._interpolation import get_xy_coords
+from esmf_regrid.experimental.unstructured_regrid import MeshInfo
+from esmf_regrid.esmf_regridder import GridInfo, Regridder
+
+
+def _get_mesh_and_dim(cube):
+    # Returns the cube's mesh and the dimension that mesh belongs to.
+    # Likely to be of the form:
+    # mesh = cube.mesh
+    # mesh_dim = cube.mesh_dim(mesh)
+    # return mesh, mesh_dim
+    pass
+
+
+def _cube_to_MeshInfo(cube):
+    # Returns a MeshInfo object describing the mesh of the cube.
+    pass
+
+
+def _cube_to_GridInfo(cube):
+    # Returns a GridInfo object describing the horizontal grid of the cube.
+    # This may be inherited from code written for the rectilinear regridding scheme.
+    pass
+
+
+def _regrid_along_dims(regridder, data, src_dim, mdtol):
+    # Before regridding, data is transposed to a standard form.
+    # This will be done either with something like the following code
+    # or else done within the regridder by specifying args.
+    new_axes = list(range(len(data.shape)))
```

Comment:
I think these are an assumption that this will be needed and what it will be like.  Maybe we can comment it out for now, and then we know it isn't part of the "definite" review here, but a reminder as to our thoughts